lib/Cake/Model/Datasource/Database/Mysql.php

```diff
@@ -895,8 +895,10 @@ public function insertMulti($table, $fields, $values) {
 				$i++;
 			}
 		}
+		$this->_connection->query('SET @OLD_SQL_MODE=@@SQL_MODE, SQL_MODE="NO_AUTO_VALUE_ON_ZERO"');
```

Comment:
I doubt Model::save() would work. If you set 'id' to 0 then model save will try to update an existing record but there is no record with id 0 yet.